modules/core/src/copy.cpp

```diff
@@ -130,31 +129,33 @@ copyMask_<ushort>(const uchar* _src, size_t sstep, const uchar* mask, size_t mst
         const ushort* src = (const ushort*)_src;
         ushort* dst = (ushort*)_dst;
         int x = 0;
-        #if CV_SSE4_2
-        if(USE_SSE4_2)//
-        {
-            __m128i zero = _mm_setzero_si128 ();
-            for( ; x <= size.width - 8; x += 8 )
+        #if CV_SIMD128
+        if( hasSIMD128()
+          #if CV_SSE4_2
+          && USE_SSE4_2
+          #endif
+        ) {
+            v_uint8x16 v_zero = v_setzero_u8();
+
+            for( ; x <= size.width - 16; x += 16 )
             {
-                 const __m128i rSrc =_mm_lddqu_si128((const __m128i*)(src+x));
-                 __m128i _mask = _mm_loadl_epi64((const __m128i*)(mask+x));
-                 _mask = _mm_unpacklo_epi8(_mask, _mask);
-                 __m128i rDst = _mm_lddqu_si128((const __m128i*)(dst+x));
-                 __m128i _negMask = _mm_cmpeq_epi8(_mask, zero);
-                 rDst = _mm_blendv_epi8(rSrc, rDst, _negMask);
-                 _mm_storeu_si128((__m128i*)(dst + x), rDst);
-             }
-        }
-        #elif CV_NEON
-        uint8x8_t v_one = vdup_n_u8(1);
-        for( ; x <= size.width - 8; x += 8 )
-        {
-            uint8x8_t v_mask = vcge_u8(vld1_u8(mask + x), v_one);
-            uint8x8x2_t v_mask2 = vzip_u8(v_mask, v_mask);
-            uint16x8_t v_mask_res = vreinterpretq_u16_u8(vcombine_u8(v_mask2.val[0], v_mask2.val[1]));
-
-            uint16x8_t v_src = vld1q_u16(src + x), v_dst = vld1q_u16(dst + x);
-            vst1q_u16(dst + x, vbslq_u16(v_mask_res, v_src, v_dst));
+                v_uint16x8 v_src1 = v_load(src + x), v_src2 = v_load(src + x + 8),
+                           v_dst1 = v_load(dst + x), v_dst2 = v_load(dst + x + 8);
+
+                v_uint8x16 v_nmask1, v_nmask2;
+                v_uint8x16 v_nmask = v_load(mask + x) == v_zero;
+                v_zip(v_nmask, v_nmask, v_nmask1, v_nmask2);
+
+                #if CV_SSE4_2
```

Comment:
unfortunately, currently we are not testing this branch; can we remove it for now?

Reply:
I did some perf tests. if it isn't enough, sure we can remove it for now.

Reply:
I believe it is better to move this branch into `v_select()` implementation directly. See [this example](https://github.com/opencv/opencv/blob/3.4.0/modules/core/include/opencv2/core/hal/intrin_sse.hpp#L1179-L1191).

Reply:
I don't think so. v_select() is a bitwise select but mask in _mm_blendv_epi8() must be packed as 8-bit of 0x00 or 0xFF.
Should I add new universal intrinsic? or just remove this branch?
Also I made a mistake here _mm_blendv_epi8() is part of SSE4.1 not SSE4.2.

Reply:
for now using `_mm_blendv_epi8()` is fine. Later on we might construct the proper universal intrinsic with the same efficiency